contracts/mailbox/src/state.rs

```diff
@@ -25,3 +27,77 @@ pub const MESSAGE_TREE: Item<MerkleTree> = Item::new(MESSAGE_TREE_KEY);
 
 pub const MESSAGE_PROCESSED_PREFIX: &str = "message_processed";
 pub const MESSAGE_PROCESSED: Map<Vec<u8>, bool> = Map::new(MESSAGE_PROCESSED_PREFIX);
+
+pub fn assert_owner(owner: Addr, sender: Addr) -> Result<(), ContractError> {
+    if owner != sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    Ok(())
+}
+
+pub fn assert_paused(storage: &dyn Storage) -> Result<(), ContractError> {
+    if !PAUSE.load(storage)? {
+        return Err(ContractError::Paused {});
+    }
+
+    Ok(())
+}
+
+pub fn assert_verify_response(resp: bool) -> Result<(), ContractError> {
+    if !resp {
+        return Err(ContractError::VerifyFailed {});
+    }
+
+    Ok(())
+}
+
+pub fn assert_addr_length(len: usize) -> Result<(), ContractError> {
+    if len > 32 {
+        return Err(ContractError::InvalidAddressLength { len });
+    }
+
+    Ok(())
+}
+
+pub fn assert_message_version(
+    message_version: u8,
+    mailbox_version: u8,
+) -> Result<(), ContractError> {
+    if message_version != mailbox_version {
+        return Err(ContractError::InvalidMessageVersion {
+            version: message_version,
+        });
+    }
+
+    Ok(())
+}
+
+pub fn assert_destination_domain(
+    message_dest_domain: u32,
+    origin_domain: u32,
+) -> Result<(), ContractError> {
+    if message_dest_domain != origin_domain {
+        return Err(ContractError::InvalidDestinationDomain {
+            domain: message_dest_domain,
+        });
+    }
+
+    Ok(())
+}
+
+pub fn assert_already_delivered(storage: &dyn Storage, id: Binary) -> Result<(), ContractError> {
+    if !MESSAGE_PROCESSED.may_load(storage, id.0.clone())?.is_none() {
```

Comment:
ask; it seems working ok with `&id.0` can you check it?

Reply:
it says when change `id.0.clone()` to `&id.0`

```
mismatched types
expected struct `std::vec::Vec<u8>`
found reference `&std::vec::Vec<u8>`
```

Reply:
oh i see